Snippets/Snippets_5/Features/FeatureConfiguration.cs

```diff
@@ -0,0 +1,72 @@
+﻿namespace Snippets5.Features
+{
+    using System;
+    using NServiceBus;
+    using NServiceBus.Features;
+    using NServiceBus.Pipeline;
+
+    class MyMagicFeature : Feature
+    {
+
+#region FeatureConfiguration
+        public MyMagicFeature()
+        {
+            EnableByDefault();
+
+            DependsOn<Sagas>();
+
+            Prerequisite(c => c.Settings.EndpointName().StartsWith("Magical"), "This feature only works on magical endpoints");
+        }
+#endregion
+
+#region FeatureSetup
+        protected override void Setup(FeatureConfigurationContext context)
+        {
+            // retrieve some global settings
+            var supervised = context.Settings.Get<bool>("SupervisorActive");
+
+            if (!supervised)
+            {
+                // register a more performant cleanup process with the container
+                context.Container.ConfigureComponent<UnsafeCleanupComponent>(DependencyLifecycle.InstancePerCall);
+            }
+
+            // add a custom behavior to the message processing pipeline which does our cleanup.
+            context.Pipeline.Register<CleanupBehaviorRegistration>();
+        }
+#endregion
+
+        void ConfigureFeatures()
+        {
+#region EnableDisableFeatures
+            var configuration = new BusConfiguration();
+            
+            // enable Saga feature since our custom feature relies on it
+            configuration.EnableFeature<Sagas>();
+
+            // this is not required if the feature uses EnableByDefault()
+            configuration.EnableFeature<MyMagicFeature>();
```

Comment:
I am :+1: for not enabling a feature from another feature (in future). Looking at the code I can see that we are using `EnableFeatureByDefault` only in parent-child feature relationship. 

If we has this parent-child (or sub-feature) relationship explicit, we could limit the enabling only to the parent where it actually makes sense.

The other flow we have with regards to dependencies is following:
 * Sagas depend on DelayedDelivery
 * DelayedDelivery depends on TimeoutManager
 * I call `config.EnableFeature<Sagas>()`
 * I call `config.DisableFeature<TimeoutManager>()`

The result is that the Sagas feature is disabled. I would argue that the desired result is an exception describing why the bus is unable to fulfill feature dependencies.